code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -188,6 +189,13 @@
 	if(method == TOUCH)
 		M.adjust_fire_stacks(-(reac_volume / 10))
 		M.ExtinguishMob()
+		if(istype(M, /mob/living/carbon))
+			var/mob/living/carbon/C = M
+			for(var/X in C.get_equipped_items())
+				var/obj/item/I = X
+				I.acid_level = 0
+				I.extinguish()
+				I.cut_overlay(GLOB.fire_overlay, TRUE)
```

Comment:
ExtinguishMob() call it. 
add I.cut_overlay(GLOB.fire_overlay, TRUE) in ExtinguishMob()
Or better just `update_icon()`

https://github.com/tgstation/tgstation/blob/85379a5269963b2463f7aa04a12e592cbe023ed3/code/modules/mob/living/carbon/carbon.dm#L853-L858